week09/exCRUD/signup_project/src/components/BoardDetail.jsx

```diff
@@ -0,0 +1,205 @@
+import React, { useState } from "react";
+import axiosInstance from "../libs/axiosInstance";
+import styled from "styled-components";
+import { useNavigate } from "react-router-dom";
+
+const Container = styled.div`
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  justify-content: center;
+  width: 100vw;
+  height: 100vh;
+`;
+
+const InputBox = styled.div`
+  margin-bottom: 1rem;
+  display: flex;
+  flex-direction: row;
+  gap: 0.5rem;
+  align-items: center;
+  button {
+    margin-bottom: 16px;
+    padding: 0.75rem 1rem;
+  }
+`;
+
+const InfoBox = styled.div`
+  margin-top: 1rem;
+  padding: 1rem;
+  border: 1px solid #ddd;
+  border-radius: 8px;
+  width: 400px;
+  background-color: #f9f9f9;
+  p {
+    color: black;
+  }
+`;
+
+const ButtonContainer = styled.div`
+  display: flex;
+  flex-direction: row;
+  justify-content: flex-end;
+  gap: 1rem;
+`;
+
+export default function BoardDetail() {
+  const navigate = useNavigate();
+  const [boardId, setBoardId] = useState("");
+  const [board, setBoard] = useState(null);
+  const [error, setError] = useState("");
+  const [editMode, setEditMode] = useState(false);
+  const [message, setMessage] = useState("");
+  const [formData, setFormData] = useState({
+    title: "",
+    description: "",
+    notice: "",
+  });
+
+  const handleFetch = async () => {
+    try {
+      const res = await axiosInstance.get(`/boards/${boardId}`);
+      setBoard(res.data);
+      setFormData({
+        title: res.data.title,
+        description: res.data.description,
+        notice: res.data.notice,
+      });
```

Comment:
요런거는 개인적으로 구조분해 할당을 하는 걸 추천드립니다!
```suggestion
const { data } =res;
const {title, description, notice} = data;
      setFormData({
        title: title,
        description:description,
        notice: notice,
      });
```